contracts/peripheral/WardQuestScheduler.sol

```diff
@@ -0,0 +1,206 @@
+
+    /**
+     * @param _cvx  Cvx token contract
+     * @param _darkQuestBoard  Dark Quest Board
+     * @param _dao  Multisig address , the owner of the contract.
+     */
+    constructor(
+        address _cvx,
+        address _darkQuestBoard,
+        address _dao
+    ) Ownable() {
+        cvx = _cvx;
+        darkQuestBoard = _darkQuestBoard;
+        _transferOwnership(_dao);
+    }
+
+    /**
+     * @notice Creates a new Quest at a predefined duration (2 epochs).
+     * @dev Creates a new Quest struct, and QuestPeriods for the Quest duration
+     * @param stash Address of the Stash linked to this Quest.
+     * @param gauge Address of the Gauge targeted by the Quest
+     * @param objective Target bias to reach (equivalent to amount of veCRV in wei to reach)
+     * @param rewardPerVote Amount of reward per veCRV (in wei)
+     * @param totalRewardAmount Total amount of rewards for the whole Quest (in wei)
+     * @param feeAmount Paladin platform fees amount (in wei)
+     * @return questID : ID of the newly created Quest
+     */
+    function createQuest(
+        address stash,
+        address gauge,
+        uint256 objective,
+        uint256 rewardPerVote,
+        uint256 totalRewardAmount,
+        uint256 feeAmount,
+        address[] calldata blacklist
+    ) external returns (uint256 questID) {
+        require(stash != address(0), "!stash");
+        // Pull all the rewards + fee in this contract
+        IERC20(cvx).safeTransferFrom(msg.sender, address(this), totalRewardAmount.add(feeAmount));
+
+        questID = IDarkQuestBoard(darkQuestBoard).createQuest(
+            gauge,
+            cvx,
+            duration,
+            objective,
+            rewardPerVote,
+            totalRewardAmount,
+            feeAmount,
+            blacklist
+        );
+        quests[questID] = stash;
+    }
+
+    /**
+     * @notice Withdraw all undistributed rewards from Closed Quest Periods and queues them to it's linked stash.
+     * @param questID ID of the Quest
+     */
+    function withdrawAndQueueUnusedRewards(uint256 questID) external returns (uint256 amount) {
+        address stash = quests[questID];
+        require(stash != address(0), "!questID");
+
+        uint256 balanceBefore = IERC20(cvx).balanceOf(address(this));
+        IDarkQuestBoard(darkQuestBoard).withdrawUnusedRewards(questID, address(this));
+        uint256 balanceAfter = IERC20(cvx).balanceOf(address(this));
+        amount = balanceAfter.sub(balanceBefore);
+        require(amount > 0, "!amount");
+
+        _queueRewards(stash, duration, amount);
+    }
+
+    /**
+     * @notice Emergency withdraws all undistributed rewards from Closed Quest Periods & all rewards for Active Periods
+     * @dev Emergency withdraws all undistributed rewards from Closed Quest Periods & all rewards for Active Periods
+     * @param questID ID of the Quest
+     */
+    function emergencyWithdraw(uint256 questID) external {
```

Comment:
If paladin kills the contract, then this function should be called. no need to protect it as it will revert in any other scenario, also the rewards are sent to the owner of the contract (multisig)